### ISSUE TITLE
ArgumentError: wrong number of arguments (given 3, expected 1..2)

### ISSUE THREAD

**Issue by MatUrbanski** [dry-rb/dry-view#146]
## Describe the bug

When I try to load the following code:

```
require "dry/view"

class ArticleView < Dry::View
  config.paths = [File.join(__dir__, "templates")]
  config.layout = "application"
  config.template = "articles/show"

  attr_reader :article_repo

  def initialize(article_repo:)
    @article_repo = article_repo
  end

  expose :article do |slug:|
    article_repo.by_slug(slug)
  end
end
```

I receive the following error:

```
ArgumentError: wrong number of arguments (given 3, expected 1..2)
from /Users/mateuszurbanski/.gem/ruby/3.0.0/gems/dry-view-0.7.0/lib/dry/view/exposures.rb:32:in `add'
```

**Comment by adam12** [comment]
This is the example from the website? How are you invoking `ArticleView`?

You'll probably need a call to `super()` inside `initialize`. Not sure if/when this broke (or if the example was never functioning to begin with). Does just adding `super()` work for you?

```ruby
def initialize(article_repo:)
  @article_repo = article_repo
  super()
end
```

**Comment by MatUrbanski** [comment]
I'm trying to use `dry-view` with Roda, and when I load this code in my IRB console it breaks.
Adding `super()` to initialize is not fixing the problem.

**Comment by MatUrbanski** [comment]
When I switched to Ruby 2.7.2 everything is working fine so it seems that this is Ruby 3.0 issue.

**Comment by adam12** [comment]
If you’re not opposed to Git dependencies, hanami/view master branch is the next release AFAIK and has many bug fixes.

**Comment by MatUrbanski** [comment]
You do not plan to update `dry-view` to be compatible with Ruby 3.0.0?

**Comment by adam12** [comment]
The `hanami/view` (master branch) is the updated `dry-view`. As part of the collaboration/unification process of Hanami and Dryrb teams, some projects moved namespaces, with `dry-view` being one of them.

**Comment by pat** [comment]
With regards to switching to `hanami/view`: how involved is the migration between the two gems? And is there any documentation around such a move?

That said: especially given there's no release out yet for that, I'd certainly appreciate a patch release of dry-view to have it 3.0-friendly, thus avoiding the need to migrate gems as part of a Ruby upgrade.

**Comment by adam12** [comment]
> With regards to switching to `hanami/view`: how involved is the migration between the two gems

I believe the current `hanami/view` master is essentially `dry-view` 0.8 + namespace change + bug fixes + Hanami 2 integration points. So you might get reasonably far with just `gsub /Dry::View/, "Hanami::View"`.

> I'd certainly appreciate a patch release of dry-view

Probably worth looking into.

**Comment by adam12** [comment]
> I'd certainly appreciate a patch release of dry-view

I submitted a PR to make specs green on `dry-view`, and requested a minor or patch to be released with the Ruby 3 fixes.

**Comment by MatUrbanski** [comment]
Thank you @adam12 👍

**Comment by MatUrbanski** [comment]
Any update on this one?

**Comment by adam12** [comment]
No, but @timriley has been insanely busy with `dry-system`. Tim, any chance you can cut a `dry-view` minor/patch?

**Comment by timriley** [comment]
Thanks for the bump, I'll get onto this today.

**Comment by MatUrbanski** [comment]
Thank you very much @timriley @adam12 👍

**Comment by timriley** [comment]
This is now released as 0.7.1.
